### PR TITLE
snap: add support for `snap advise-snap pkgName`

### PR DESCRIPTION
**Pull request by mvo5**
This implements `snap advise-snap <pkgName>` and will give useful
advise if the given pkgname is available in the store.

Based on https://github.com/snapcore/snapd/pull/4456

### CODE REVIEW

**Comment by chipaca** [comment]
advi**s**e

File: cmd/snap/cmd_advise.go [hunk 1]
```diff
@@ -29,25 +29,29 @@ import (
...
-	Format string `long:"format" default:"pretty"`
+	Format  string `long:"format" default:"pretty"`
+	Command bool   `long:"command"`
 }
 
-var shortAdviseCommandHelp = i18n.G("Advise on available snaps.")
-var longAdviseCommandHelp = i18n.G(`
+var shortAdviseSnapHelp = i18n.G("Advise on available snaps.")
+var longAdviseSnapHelp = i18n.G(`
 The advise-command command shows what snaps with the given command are
```
Comment (chipaca) [inline cmd/snap/cmd_advise.go]: missed this one

**Comment by mvo5** [comment]
@chipaca this PR should take care of everything (including storing the summary).

File: cmd/snap/main.go [hunk 2]
```diff
@@ -279,7 +279,21 @@ func resolveApp(snapApp string) (string, error) {
 func main() {
 	cmd.ExecInCoreSnap()
 
-	// magic \o/
+	// check for magic symlink to /usr/bin/snap:
+	// 1. symlink from command-not-found to /usr/bin/snap: run c-n-f
+	if os.Args[0] == filepath.Join(dirs.GlobalRootDir, "/usr/lib/command-not-found") {
```
Comment (chipaca) [inline cmd/snap/main.go]: <3

File: store/store_test.go [hunk 3]
```diff
@@ -2688,7 +2688,7 @@ const mockNamesJSON = `
         "package_name": "foo"
       }
     ]
-  }
+q  }
```
Comment (chipaca) [inline store/store_test.go]: typo?

File: advisor/backend.go [hunk 4]
```diff
@@ -28,18 +28,22 @@ import (
 	"github.com/snapcore/snapd/dirs"
 )
 
-var cmdBucketKey = []byte("Commands")
+var (
+	cmdBucketKey = []byte("Commands")
+	pkgBucketKey = []byte("Pkgs")
```
Comment (chipaca) [inline advisor/backend.go]: if I had a nit, it'd be that I'd've gone with `"Snaps"` here

File: store/store_test.go (hunk 3 above)
Comment (mvo5) [reply store/store_test.go]: Yeah, fat fingers! Thanks for noticing.

File: advisor/backend.go [hunk 5]
```diff
@@ -206,3 +227,25 @@ func (f *boltFinder) Find(command string) ([]Command, error) {
...
+	}
+
+	bsummary := b.Get([]byte(pkgName))
+	if bsummary == nil {
+		return nil, nil
+	}
+
+	pkg := Package{Snap: pkgName, Summary: string(bsummary)}
+
+	return []Package{pkg}, nil
```
Comment (bboozzoo) [inline advisor/backend.go]: The function is declared to return a list of pacakges, but in fact it returns only one. Is there a plan to do partial matches on package name?

Comment (zyga) [reply advisor/backend.go]: I’ll add a small request to just document the function as a part of the answer

Comment (mvo5) [reply advisor/backend.go]: Good point, this is for exact matches only, fixed now.

**Pull request merged**